dist/smoothscroll.js

```diff
@@ -95,11 +95,23 @@
      * @returns {Node} el
      */
     function findScrollableParent(el) {
+      var isBody;
+      var hasScrollableSpace;
+      var hasVisibleOverflow;
+
       do {
         el = el.parentNode;
-      } while (el !== d.body
-              && !(el.clientHeight < el.scrollHeight
-              || el.clientWidth < el.scrollWidth));
+
+        // set condition variables
+        isBody = el === d.body;
+        hasScrollableSpace =
+          el.clientHeight < el.scrollHeight ||
+          el.clientWidth < el.scrollWidth;
+        hasVisibleOverflow =
+          w.getComputedStyle(el, null).overflow === 'visible';
+      } while (!isBody && !(hasScrollableSpace && !hasVisibleOverflow));
```

Comment:
Bare with me and this not so clear statement. Script will keep looking for a parent to scrollable unless the element is the `body` element or has space to scroll but `overflow` is not set to `visible`. I hope that simplifies the reading.